### PR TITLE
Bug/widgets transistion feels delayed

### PR DESCRIPTION
**Pull request by lily-law**

### CODE REVIEW

**Comment by yanpelov** [comment]
1. the dashnav doesn't seem to work quite well when less than 5 items.
i think the dashnav should shorten dynamically and look & feel like when we have 5 items.
i would suggest changing the dashnav slots number to 3 when (4 - 3 dashes) and 1 slot for (2 - 1 dashes).
currently, the missing dashes just appear as a blank space and when traversing all the dashes, it jumps inconsistently between the last and the first dash.
2. when traversing to the active dash, there is a slight scroll jump, that moves the screen a few pixels below the edge of the dashnav and when you go to the next dash, it jump backs.
3. when adding new dash, i would suggest adding an 'Enter' keypress to accept the user input, rather then typing and clicking somewhere on the screen to accept the input. i believe it's more intuitive, as you expect to press Enter after typing
4. upon first load, the positioning of the Delete Widget is not symmetrical to the Add Widget.
5. upon load, the first focus is on the text of the Delete Widget, and after traversing dashes it changes to the Welcome widget. I would suggest removing the focusing on the text on any widget upon load.
6. if i change some widget positioning, and i traverse through the dashes, it jumps below the dash nav upon active dash and jumps back when going to an empty dash
7. i would suggest holding with new features (like markup) until clearing out all the issues, as it creates new code that needs to be checked and it's all compressed in one huge PR which is hard to follow...

**Comment by lily-law** [comment]
Hi yanpelov, we haven't got a lot of time left on this voyage, I'm thinking the Dashnav needs to be less a priority ATM, we can always revert to a simple/basic dash navigation menu if it has too many issues and we run out of time. Right now it's more important that we get enough basic functionality built in so the app can be useful to users. A big issue right now is we need the data store to push and pull from the browsers local storage API and to get the Sticky Widgets to offer a little more convenience to users when entering data.

I think we could work in some quick fixes for the points you've raised: 
1. I think for less then five items we just need to disable clicking forwards at the end of list and visa-versa
2. The scroll jump is proving to be tricky, I'm wondering if maybe we need to either make each item have a fixed width (longer text titles would have to shrink) or change to JS animations so we can slide the carousel by a dynamic amount depending on the size of each item
3. Yes totally putting blurring input on enter key will be worth while, also I had in mind later of putting in swipe inputs for switching between dashes
4. I'm not seeing that, could you do a screenshot?
5. In the branch I'm working on now feature/Sticky I've removed the autofocus. We put them in initially because when textarea was clicked and you began typing it lost focus when detect input assigned the Sticky to a type (Text). In my new branch I've changed it so all editing happens in a base Sticky Widget and detectInput only assigns after editing is finished. Only downside is It takes two clicks after, one to enter edit mode, two to focus the input
6. Re-positioning/saving positioning of widigets is not yet implemented, we had a save widget position function but because it didn't account for other widgets moving and different screen sizes it wasn't consistent. For this I think we need to implement something that snapshots the positions of all widgets on each re-position/resize with reference to the current number of columns
7. We were amounting a long list of open PRs, maybe it was fine and I should leave them open so we can see the progress clearer? I don't believe we can make the code perfect before merging and finish the project in time, being as small a team as we are and with the hours we're able to spend on this. I think it's better we have some bugs/imperfect code and get the app finished then having ultra refined code for a non functional app
